### PR TITLE
Fix Link That Halt Link Validation Tooling and Links to non-Java SDK Repos

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

This PR fixes a link that prevents the link validation tooling from continuing to run (link never responds to the tooling and it deadlocks) and links that point to non-Java SDK repos.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/modelsrepository/azure-iot-modelsrepository/src/samples/README.md [hunk 1]
```diff
@@ -196,13 +196,8 @@ System.out.println(fullyQualifiedModelUri);
 
 <!-- LINKS -->
 [modelsrepository_github_repo]: https://github.com/Azure/iot-plugandplay-models
-[modelsrepository_sample_extension]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/modelsrepository/Azure.IoT.ModelsRepository/samples/ModelsRepositoryClientSamples/ModelsRepositoryClientExtensions.cs
-[modelsrepository_clientoptions]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/modelsrepository/Azure.IoT.ModelsRepository/src/ModelsRepositoryClientOptions.cs
```
Comment (alzimmermsft) [inline sdk/modelsrepository/azure-iot-modelsrepository/src/samples/README.md:200]: These weren't being used and point to the .NET repo

File: sdk/modelsrepository/azure-iot-modelsrepository/src/samples/README.md [hunk 2]
```diff
@@ -196,13 +196,8 @@ System.out.println(fullyQualifiedModelUri);
...
 [modelsrepository_msdocs]: https://docs.microsoft.com/azure/iot-pnp/concepts-model-repository
 [modelsrepository_publish_msdocs]: https://docs.microsoft.com/azure/iot-pnp/concepts-model-repository#publish-a-model
 [modelsrepository_iot_endpoint]: https://devicemodels.azure.com/
 [json_ld_reference]: https://json-ld.org
 [dtdlv2_reference]: https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md
-[azure_core_transport]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Pipeline.md
-[azure_core_diagnostics]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md
-[azure_core_configuration]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Configuration.md
```
Comment (alzimmermsft) [inline sdk/modelsrepository/azure-iot-modelsrepository/src/samples/README.md:208]: These weren't being used and point to the .NET repo

File: sdk/quantum/azure-quantum-jobs/README.md [hunk 3]
```diff
@@ -198,7 +198,7 @@ additional questions or comments.
 [workspaces]: https://docs.microsoft.com/azure/quantum/how-to-create-quantum-workspaces-with-the-azure-portal
 [location]: https://azure.microsoft.com/global-infrastructure/services/?products=quantum
 [blob-storage]: https://docs.microsoft.com/azure/storage/blobs/storage-blobs-introduction
-[contributing]: https://github.com/Azure/azure-sdk-for-net/tree/main/CONTRIBUTING.md
+[contributing]: https://github.com/Azure/azure-sdk-for-java/tree/main/CONTRIBUTING.md
```
Comment (alzimmermsft) [inline sdk/quantum/azure-quantum-jobs/README.md:201]: Should point to the Java repo

File: sdk/remoterendering/test-resources-post.ps1 [hunk 4]
```diff
@@ -2,8 +2,8 @@
 # Licensed under the MIT License.
 
 # This script is used to generate the Test Configuration file for Storage live tests.
-# It is invoked by the https://github.com/Azure/azure-sdk-for-net/blob/main/eng/New-TestResources.ps1
-# script after the ARM template, defined in https://github.com/Azure/azure-sdk-for-net/blob/arm-template-storage/sdk/storage/test-resources.json,
+# It is invoked by the https://github.com/Azure/azure-sdk-for-java/blob/main/eng/New-TestResources.ps1
+# script after the ARM template, defined in https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/remoterendering/test-resources.json,
```
Comment (alzimmermsft) [inline sdk/remoterendering/test-resources-post.ps1:6]: Should point to the Java repo and the second link doesn't exist either

File: sdk/remoterendering/test-resources-post.ps1 [hunk 5]
```diff
@@ -2,8 +2,8 @@
 # Licensed under the MIT License.
 
 # This script is used to generate the Test Configuration file for Storage live tests.
-# It is invoked by the https://github.com/Azure/azure-sdk-for-net/blob/main/eng/New-TestResources.ps1
-# script after the ARM template, defined in https://github.com/Azure/azure-sdk-for-net/blob/arm-template-storage/sdk/storage/test-resources.json,
```
Comment (alzimmermsft) [inline sdk/remoterendering/test-resources-post.ps1:6]: Should point to the Java repo and the second link doesn't exist either

**Review by JonathanGiles — APPROVED** [review]

**Pull request merged**